### PR TITLE
Add count param

### LINKED ISSUE #1119: [Sync Optimization] Update all API call to use a count of 100
**Description of feature**
- The current API `_count` when querying the server is 20.  This takes Approx 7mins to sync.  
- Increasing the count to 100 increases the amount of data we receive per call and reduces the sync period to approx 3 mins

**Describe the enhancement**
- [x] Update all FHIR core calls to the API to default to a 100 count. 
- [x] Make the count definition configurable, this will enable other implementations to change their count settings if need be.  This can be defined via configs.  
     - [x]  If it's not defined then the default is 100
     - [x] If defined the default becomes the new number set on the configs. 

- We can use the [application configuration](https://github.com/opensrp/fhircore/blob/main/android/anc/src/main/assets/configurations/app/application_configurations.json) to hold the count config

@pld @maimoonak What do you think about making the count definition configurable.

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1119 #949

**Type**
Choose one: Feature

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by FikriMilano** [comment]
@maimoonak  I can't run the app after merging the latest changes, even in the main branch.
The app crashes after submitting the app config name. Either Quest or ANC.

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt:58]: let's move these timeouts to constants

**Review by pld — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt
Comment (FikriMilano) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/di/NetworkModule.kt:58]: Okay

**Review by ellykits — APPROVED** [review]

**Comment by pld** [comment]
> @pld could you approve for this changes? (couldn't merge without it)

apologies for the delay, I've dismissed my stale review, you should be all set now!

**Pull request merged**
